src/main/java/org/springframework/data/tarantool/core/TarantoolTemplate.java

```diff
@@ -88,6 +103,81 @@ public TarantoolTemplate(
         );
     }
 
+    private MessagePackMapper getDefaultComplexTypesMapper() {
+        // blocked because it's not deepcopy
+        // MessagePackMapper defaultConverterWithTupleConverter = tarantoolClient.getConfig().getMessagePackMapper().copy();
+        return DefaultMessagePackMapperFactory
```

Comment:
I agree that the config from the client should remain. But to use now, we have to register one `TarantoolAutoResultConverter` with all the metadata. With each request, we cannot register a new converter.

> since you aren't going to change the inner mapper stack (returned from config), you just add wrappers around and for them take care of the proper copying

I didn't understand what kind of wrappers you mean, please describe in more detail

Reply:
Discussed with @vrogach2020 that we should leave it as it is, not reuse the custom one until we fix this problem in [cartridge-java#166](https://github.com/tarantool/cartridge-java/issues/166).